pkg/local_workflows/data_transformation_workflow.go

```diff
@@ -0,0 +1,108 @@
+package localworkflows
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"cuelang.org/go/cue/cuecontext"
+	cuejson "cuelang.org/go/pkg/encoding/json"
+	cueutil "github.com/snyk/go-application-framework/internal/cueutils"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	DataTransformationWorkflowName = "datatransformation"
+)
+
+var WORKFLOWID_DATATRANSFORMATION = workflow.NewWorkflowIdentifier(DataTransformationWorkflowName)
+
+func InitDataTransformationWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(DataTransformationWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_DATATRANSFORMATION, workflow.ConfigurationOptionsFromFlagset(flags), dataTransformationEntryPoint)
+
+	engine.GetConfiguration().AddDefaultValue(configuration.FF_TRANSFORMATION_WORKFLOW, configuration.StandardDefaultValueFunction(false))
+	return err
+}
+
+func dataTransformationEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	ff_transform_enabled := config.GetBool(configuration.FF_TRANSFORMATION_WORKFLOW)
+	output = input
+
+	logger.Println("dataTransformation workflow start")
+
+	if !ff_transform_enabled {
+		return output, nil
+	}
+
+	for _, data := range input {
+		if strings.HasPrefix(data.GetContentType(), content_type.SARIF_JSON) {
+			// process input
+			d, err := transformSarifData(data)
+			if err != nil {
+				return output, err
+			}
+
+			bytes, err := json.Marshal(d)
+			if err != nil {
+				return output, err
+			}
+
+			output = append(input, workflow.NewData(
+				workflow.NewTypeIdentifier(WORKFLOWID_DATATRANSFORMATION, DataTransformationWorkflowName),
+				content_type.LOCAL_FINDING_MODEL,
+				bytes, workflow.WithConfiguration(config), workflow.WithLogger(logger)))
+		}
+	}
+
+	return output, nil
+}
+
+func transformSarifData(singleData workflow.Data) (result local_models.LocalFinding, err error) {
+	var localFinding local_models.LocalFinding
+	jsonData, ok := singleData.GetPayload().([]byte)
+	if !ok {
+		return localFinding, err
```

Comment:
This pattern is based on the following:
> yields an additional untyped boolean value. The value of ok is true if the key x is present in the map, and false otherwise.
Source: https://go.dev/ref/spec#Type_identity